sw/device/tests/crt_test.c

```diff
@@ -54,6 +58,91 @@ static void init_uart(void) {
   base_uart_stdout(&uart0);
 }
 
+static void test_crt_section_clear(void) {
+  // Maximum end index of target section.
+  const size_t len = 32;
+
+  // Section indices (start inclusive, end exclusive).
+  struct {
+    size_t start;
+    size_t end;
+  } tests[6] = {{.start = 0, .end = 0},         {.start = 0, .end = 1},
+                {.start = len - 1, .end = len}, {.start = 0, .end = len - 1},
+                {.start = 1, .end = len},       {.start = 0, .end = len}};
+
+  for (size_t t = 0; t < sizeof(tests) / sizeof(tests[0]); ++t) {
+    // Set target array to non-zero values.
+    uint32_t section[len];
+    const uint32_t val = ~0u;
+    for (size_t i = 0; i < len; ++i) {
+      section[i] = val;
+    }
+
+    // Clear section of target array.
+    const size_t start = tests[t].start;
+    const size_t end = tests[t].end;
+    crt_section_clear(&section[start], &section[end]);
+
+    // Check that section was cleared.
+    for (size_t i = 0; i < len; ++i) {
+      uint32_t expect = i >= start && i < end ? 0 : val;
+      CHECK(section[i] == expect,
+            "%s case %u: section[%u] got 0x%08x, want 0x%08x", __func__, t, i,
+            section[i], expect);
+    }
+  }
+}
+
+static void test_crt_section_copy(void) {
+  // Maximum end index of target section.
+  const size_t len = 32;
+
+  // Section indices (start inclusive, end exclusive) and source index
+  // (inclusive).
+  struct {
+    size_t start;
+    size_t end;
+    size_t source;
+  } tests[14] = {{.start = 0, .end = 0, .source = 0},
+                 {.start = 0, .end = 1, .source = 1},
+                 {.start = len - 1, .end = len, .source = 2},
+                 {.start = 0, .end = len - 1, .source = 1},
+                 {.start = 1, .end = len, .source = 1},
+                 {.start = 0, .end = len, .source = 0},
+                 {.start = 0, .end = len, .source = 0},
+                 {.start = 1, .end = len, .source = 0},
+                 {.start = 2, .end = len, .source = 0},
+                 {.start = 3, .end = len, .source = 0},
+                 {.start = 0, .end = len / 2, .source = 0},
+                 {.start = 1, .end = len / 2, .source = 0},
+                 {.start = 2, .end = len / 2, .source = 0},
+                 {.start = 3, .end = len / 2, .source = 0}};
+
+  for (size_t t = 0; t < sizeof(tests) / sizeof(tests[0]); ++t) {
+    // Clear target array and setup source array with known values.
+    uint32_t dst[len], src[len], gold[len];
```

Comment:
wouldn't it make more sense to have a global
```
static const uint32_t kGold[32] = { ... };
```
and then write this in terms of `ARRAYSIZE(kGold)` or such? Could also do this in the clearing example.

Reply:
I ended up getting rid of the gold array and just using a calculated value instead.